### PR TITLE
[bitnami/keycloak] Fix NetworkPolicy to open the Infinispan port (7800)

### PR DESCRIPTION
**Pull request by agtogna**
### Description of the change

Open port 7800 in the NetworkPolicy to allow Infinispan cache traffic between pods.

### Benefits

Without port 7800 the pods Infinispan caches do not see each other, thus having a split cache (equivalent to local cache)

### Applicable issues

#14575 

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by Mauraza** [comment]
Hi @agtogna,

Thanks for the contribution!!
Did you try to add to the port in the [statefulset](ttps://github.com/bitnami/charts/blob/main/bitnami/keycloak/templates/statefulset.yaml) instead of the NetworkPolicy?

**Comment by agtogna** [comment]
> Hi @agtogna,
> 
> Thanks for the contribution!! Did you try to add to the port in the statefulset instead of the NetworkPolicy?

Hello @Mauraza , I did not try it, however I don't think it would work since the 7800 port would still not be in the networkPolicy

**Comment by Mauraza** [comment]
> Hello @Mauraza , I did not try it, however I don't think it would work since the 7800 port would still not be in the networkPolicy

But I think this needs to be in the statefulset.yaml too. 
Could you test if is working to add these values 🔽  to the ones you already have?
```yaml
extraContainerPorts: 
  - name: Infinispan
    containerPort: 7800
    protocol: TCP
service:
  extraPorts: 
    - name: Infinispan
      port: 7800
      targetPort: Infinispan
      protocol: TCP 
```

**Comment by agtogna** [comment]
> 

Hello @Mauraza , I've tried adding both, but I still need to modify the network policy to allow traffic on port 7800/TCP.
By the way, I do not think adding the 7800 in the Service is a good idea: it should not be reachable as a service, should only be used by the pods to communicate between them.
I've only added the `extraContainerPorts` to my PR.

File: bitnami/keycloak/values.yaml [hunk 1]
```diff
@@ -251,7 +251,10 @@ containerPorts:
   https: 8443
 ## @param extraContainerPorts Optionally specify extra list of additional port-mappings for Keycloak container
 ##
-extraContainerPorts: []
+extraContainerPorts: 
+  - name: infinispan
+    containerPort: 7800
+    protocol: TCP
```
Comment (Mauraza) [inline bitnami/keycloak/values.yaml:257]: This is not necessary to change. 
```suggestion
extraContainerPorts: []
```
You must add this part to the container port:
```yaml
containerPorts:
  http: 8080
  https: 8443
  infinispan: 7800
``` 
And in the satefulset.yaml:
```yaml
   ...
          ports:
          ...
            - name: infinispan
              containerPort: {{ .Values.containerPorts.infinispan }}
              protocol: TCP
...
```

File: bitnami/keycloak/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -17,6 +17,7 @@ spec:
     matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
   ingress:
     - ports:
+        - port: 7800
```
Comment (Mauraza) [inline bitnami/keycloak/templates/networkpolicy.yaml:20]:
```suggestion
        - port: {{ .Values.containerPorts.infinispan }}
```

**Review by Mauraza — CHANGES_REQUESTED** [review]
Hi @agtogna,

Could you please check my comments?

File: bitnami/keycloak/Chart.yaml [hunk 3]
```diff
@@ -26,4 +26,4 @@ name: keycloak
 sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/keycloak
   - https://github.com/keycloak/keycloak
-version: 13.0.3
+version: 13.0.4
```
Comment (Mauraza) [inline bitnami/keycloak/Chart.yaml:29]:
```suggestion
version: 13.1.0
```

**Review by Mauraza — CHANGES_REQUESTED** [review]

File: bitnami/keycloak/values.yaml [hunk 4]
```diff
@@ -249,6 +249,7 @@ replicaCount: 1
 containerPorts:
```
Comment (Mauraza) [inline bitnami/keycloak/values.yaml:249]: The only this missing is updating the comments with the new port:
```yaml
## @param containerPorts.http Keycloak HTTP container port
## @param containerPorts.https Keycloak HTTPS container port
## @param containerPorts.infinispan Keycloak infinispan container port
```

**Review by Mauraza — COMMENTED** [review]
Hi @agtogna,

The tests are failing for a little thing, could you update values.yaml?

**Comment by Mauraza** [comment]
Hi @agtogna,

I missing port infinispan (7800) in statefulset.yaml
```yaml
.....
          ports:
            - name: http
              containerPort: {{ .Values.containerPorts.http }}
              protocol: TCP
            {{- if .Values.tls.enabled }}
            - name: https
              containerPort: {{ .Values.containerPorts.https }}
              protocol: TCP
            {{- end }}
            - name: infinispan
              containerPort: {{ .Values.containerPorts.infinispan }}
              protocol: TCP
....
```

File: bitnami/keycloak/README.md [hunk 5]
```diff
@@ -330,14 +341,15 @@ The command removes all the Kubernetes components associated with the chart and
...
+
 ### Keycloak Logging parameters
 
 | Name             | Description                                                                    | Value     |
 | ---------------- | ------------------------------------------------------------------------------ | --------- |
 | `logging.output` | Alternates between the default log output format or json format                | `default` |
 | `logging.level`  | Allowed values as documented: FATAL, ERROR, WARN, INFO, DEBUG, TRACE, ALL, OFF | `INFO`    |
 
-Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
+
```
Comment (Mauraza) [inline bitnami/keycloak/README.md:352]: Sorry, Could you add this line? 
```suggestion
Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example,
```

**Review by Mauraza — APPROVED** [review]
Hi @agtogna,

Thanks for the contribution! LGTM!

**Pull request merged**